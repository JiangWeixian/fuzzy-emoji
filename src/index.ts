import Fuzzysearch from 'fuzzy-search'
import { gemoji } from 'gemoji'

export const fuzzysearch = new Fuzzysearch(gemoji, ['names', 'tags', 'descriptions'], {
  sort: true,
})
