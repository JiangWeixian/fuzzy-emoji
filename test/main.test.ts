import { fuzzysearch } from '../src'

describe('fuzzysearch', () => {
  test('search', () => {
    expect(fuzzysearch.search('love').slice(0, 10)).toMatchSnapshot()
  })
})
