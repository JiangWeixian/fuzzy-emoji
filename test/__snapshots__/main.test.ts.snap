// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`fuzzysearch search 1`] = `
Array [
  Object {
    "category": "Smileys & Emotion",
    "description": "smiling face with hearts",
    "emoji": "🥰",
    "names": Array [
      "smiling_face_with_three_hearts",
    ],
    "tags": Array [
      "love",
    ],
  },
  Object {
    "category": "Smileys & Emotion",
    "description": "smiling face with heart-eyes",
    "emoji": "😍",
    "names": Array [
      "heart_eyes",
    ],
    "tags": Array [
      "love",
      "crush",
    ],
  },
  Object {
    "category": "Smileys & Emotion",
    "description": "heart with arrow",
    "emoji": "💘",
    "names": Array [
      "cupid",
    ],
    "tags": Array [
      "love",
      "heart",
    ],
  },
  Object {
    "category": "Smileys & Emotion",
    "description": "red heart",
    "emoji": "❤️",
    "names": Array [
      "heart",
    ],
    "tags": Array [
      "love",
    ],
  },
  Object {
    "category": "Smileys & Emotion",
    "description": "love letter",
    "emoji": "💌",
    "names": Array [
      "love_letter",
    ],
    "tags": Array [
      "email",
      "envelope",
    ],
  },
  Object {
    "category": "People & Body",
    "description": "love-you gesture",
    "emoji": "🤟",
    "names": Array [
      "love_you_gesture",
    ],
    "tags": Array [],
  },
  Object {
    "category": "Travel & Places",
    "description": "love hotel",
    "emoji": "🏩",
    "names": Array [
      "love_hotel",
    ],
    "tags": Array [],
  },
  Object {
    "category": "Activities",
    "description": "boxing glove",
    "emoji": "🥊",
    "names": Array [
      "boxing_glove",
    ],
    "tags": Array [],
  },
  Object {
    "category": "Objects",
    "description": "gloves",
    "emoji": "🧤",
    "names": Array [
      "gloves",
    ],
    "tags": Array [],
  },
  Object {
    "category": "Flags",
    "description": "flag: Slovenia",
    "emoji": "🇸🇮",
    "names": Array [
      "slovenia",
    ],
    "tags": Array [],
  },
]
`;
